File: src/types/analysis.ts
export interface AnalysisResult {
  visualSimilarity: {
    score: number;
    segments: SegmentAnalysis[];
  };
  textAnalysis: {
    score: number;
    inconsistencies: TextInconsistency[];
  };
  handwritingAnalysis: {
    score: number;
    anomalies: HandwritingAnomaly[];
  };
  overallScore: number;
}

export interface SegmentAnalysis {
  id: string;
  region: string;
  similarityScore: number;
  issues?: string[];
}

export interface TextInconsistency {
  location: string;
  type: 'semantic' | 'structural' | 'linguistic';
  description: string;
  severity: 'low' | 'medium' | 'high';
}

export interface HandwritingAnomaly {
  location: string;
  type: 'style' | 'pressure' | 'spacing';
  confidence: number;
  description: string;
}

File: backend/src/models/analysis.py
from pydantic import BaseModel
from typing import List, Literal

class SegmentAnalysis(BaseModel):
    id: str
    region: str
    similarity_score: float
    issues: List[str] = []

class TextInconsistency(BaseModel):
    location: str
    type: Literal["semantic", "structural", "linguistic"]
    description: str
    severity: Literal["low", "medium", "high"]

class HandwritingAnomaly(BaseModel):
    location: str
    type: Literal["style", "pressure", "spacing"]
    confidence: float
    description: str

class VisualSimilarity(BaseModel):
    score: float
    segments: List[SegmentAnalysis]

class TextAnalysis(BaseModel):
    score: float
    inconsistencies: List[TextInconsistency]

class HandwritingAnalysis(BaseModel):
    score: float
    anomalies: List[HandwritingAnomaly]

class AnalysisResult(BaseModel):
    visual_similarity: VisualSimilarity
    text_analysis: TextAnalysis
    handwriting_analysis: HandwritingAnalysis
    overall_score: float